src/ensemble/ensemble_problems.jl

```diff
@@ -44,6 +44,26 @@ function EnsembleProblem(; prob,
     EnsembleProblem(prob; prob_func, output_func, reduction, u_init, safetycopy)
 end
 
+function EnsembleProblem(; prob,
+    u0s::Union{Nothing, Vector{uType}} = nothing,
+    prob_func = (prob, i, repeat) -> remake(prob, u0 = u0s[i]),
+    output_func = DEFAULT_OUTPUT_FUNC,
+    reduction = DEFAULT_REDUCTION,
+    u_init = nothing, p = nothing,
+    safetycopy = prob_func !== DEFAULT_PROB_FUNC) where {uType}
+    EnsembleProblem(prob; prob_func, output_func, reduction, u_init, safetycopy)
+end
+
+function EnsembleProblem(; prob,
+    trajectories::Int,
+    prob_func,
```

Comment:
this doesn't define a prob_func?

Reply:
Yes, because it'll need to access generated u0s, like https://github.com/SciML/Optimization.jl/pull/620/files#diff-6eea19b562ee1839438c6fa37d876324d501594af403caf5c836620ad7e3b793R6 or as a global variable so